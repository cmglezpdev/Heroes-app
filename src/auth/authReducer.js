import { types } from "../types/types";

// const state = {
//     name: 'Carlos',
//     logged: true
// }


export const authReducer = ( state = {}, action ) => {

    switch( action.type ) {
        case types.login:
            return {
                ...action.payload,
                logged: true
            }

        case types.logout:
            return {
                logged: false
            }

        default:
            return state;
    }
}