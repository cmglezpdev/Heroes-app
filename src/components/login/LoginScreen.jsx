
export const LoginScreen = () => {
  return (
    <div>LoginScreen</div>
  )
}
