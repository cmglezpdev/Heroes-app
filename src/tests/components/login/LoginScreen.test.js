import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { AuthContext } from '../../../auth/authContext';
import { LoginScreen } from '../../../components/login/LoginScreen';
import { types } from '../../../types/types';

const mockDispatch = jest.fn();
const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate,
}))

describe('Tests about LoginScreen', () => {

    test('should show it correctly', () => {

        const contextValue = {
            user: {
                name: 'Carlos Manuel',
                logged: true
            },
            dispatch: () => {}
        }

        const{ container } = render(
            <AuthContext.Provider value={contextValue}>
                <MemoryRouter initialEntries={['/login']}>
                    <LoginScreen />
                </MemoryRouter>
            </AuthContext.Provider>
        );

        expect( container ).toMatchSnapshot();
    })

    test('should call the dispatch and to do navigation', () => {
        const contextValue = {
            user: {
                name: 'Carlos Manuel',
                logged: true
            },
            dispatch: mockDispatch
        }

        render(
            <AuthContext.Provider value={contextValue}>
                <MemoryRouter initialEntries={['/login']}>
                    <LoginScreen />
                </MemoryRouter>
            </AuthContext.Provider>
        );

        const button = screen.getByRole('button');
        fireEvent.click(button);

        const action = {
            type: types.login,
            payload: {name: 'Carlos Manuel'}
        };

        expect( mockDispatch ).toHaveBeenCalledTimes(1);
        expect( mockDispatch ).toHaveBeenCalledWith(action);
    
        const lastPath = localStorage.getItem("lastPath") || "/";
        expect( mockNavigate ).toHaveBeenCalledTimes(1);
        expect( mockNavigate ).toHaveBeenCalledWith(lastPath, { replace: true }) 
    })

})