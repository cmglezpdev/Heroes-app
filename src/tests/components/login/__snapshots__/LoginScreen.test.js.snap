// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests about LoginScreen should show it correctly 1`] = `
<div>
  <div
    class="container mt-5"
  >
    <h1>
      Login
    </h1>
    <hr />
    <button
      class="btn btn-primary"
    >
      Login
    </button>
  </div>
</div>
`;
